lib/urn.js

```diff
@@ -210,8 +213,8 @@ module.exports = {
         var schema = encoded.slice(nextIndex + 1 * 2, nextIndex + (1 + schemaLength) * 2);
         // Convert to string
         schema = hexToString(schema);
-        var id = encoded.slice(nextIndex + (1 + schemaLength + 2) * 2, nextIndex + (1 + schemaLength + 2 + idLength) * 2);
-
+        var idHex = encoded.slice(nextIndex + (1 + schemaLength + 2) * 2, nextIndex + (1 + schemaLength + 2 + idLength) * 2);
+        var id = RECOGNIZED_PROTOCOLS.has(schema) ? idHex : hexToString(idHex);
```

Comment:
Just a note. We need to make sure that when we add new "recognized protocol" we haven't used it with any of the things before (it will result in diff output). So if we decided to recognize `mycategory:1.0:1237197`, we have to change something in category, like version name or sub-category, say `mycategory:1.0:sub:1237197` to not interfere with existing items.